weaviate/collections/batch/grpc_batch_objects.py

```diff
@@ -247,53 +247,53 @@ def __translate_properties_from_python_to_grpc(
         object_properties: List[base_pb2.ObjectProperties] = []
         object_array_properties: List[base_pb2.ObjectArrayProperties] = []
 
-        for key, val in refs.items():
-            if isinstance(val, _Reference):
-                if val.is_multi_target:
+        for key, ref in refs.items():
+            if isinstance(ref, _Reference):
+                if ref.is_multi_target:
                     multi_target.append(
                         batch_pb2.BatchObject.MultiTargetRefProps(
-                            uuids=val.uuids_str,
-                            target_collection=val.target_collection,
+                            uuids=ref.uuids_str,
+                            target_collection=ref.target_collection,
                             prop_name=key,
                         )
                     )
                 else:
                     single_target.append(
                         batch_pb2.BatchObject.SingleTargetRefProps(
-                            uuids=val.uuids_str, prop_name=key
+                            uuids=ref.uuids_str, prop_name=key
                         )
                     )
-            elif isinstance(val, str):
+            elif isinstance(ref, str) or isinstance(ref, uuid_package.UUID):
                 single_target.append(
-                    batch_pb2.BatchObject.SingleTargetRefProps(uuids=[val], prop_name=key)
+                    batch_pb2.BatchObject.SingleTargetRefProps(uuids=[str(ref)], prop_name=key)
                 )
-            elif isinstance(val, uuid_package.UUID):
+            elif isinstance(ref, list):
                 single_target.append(
                     batch_pb2.BatchObject.SingleTargetRefProps(
-                        uuids=[str(object=val)], prop_name=key
+                        uuids=[str(v) for v in ref], prop_name=key
                     )
                 )
             else:
-                raise WeaviateInvalidInputError(f"Invalid reference: {val}")
+                raise WeaviateInvalidInputError(f"Invalid reference: {ref}")
 
-        for key, val in data.items():
-            if isinstance(val, _Reference):
-                if val.is_multi_target:
+        for key, datum in data.items():
```

Comment:
why this rename? (In German `datum`means date so I find it a bit confusing)

Reply:
Oh really? `datum` is the singular of `data`, which originally comes from Latin. I made the change because of `mypy` problems due to `val` being shared between `refs.items()` and `data.items()`. I will change `datum` to `entry` here though, no problem!

Reply:
thank you!